proposals/1681/naming_schemas.md

```diff
@@ -0,0 +1,90 @@
+---
+Author: Benji Lampel
+Created: 03/07/2023
+Issue: https://github.com/OpenLineage/OpenLineage/issues/1681
+---
+
+**Purpose**
+The Naming.md file should be reworked as a more programmatic solution with clear, specific definitions.
+
+Names and Namespaces are currently slightly nebulous concepts. They can best be described, perhaps, by two components that, together, form a unique URI to a specific dataset for datasets, and a unique name for a Job. This is a serviceable definition for databases, datalakes, and distributed file stores, where a well-defined path exists by nature of the structure. But this may not be the case with database-like systems, or non-database systems, like Salesforce or Google Sheets, respectively. These instances need another way to specify a unique name, one that may not resolve to a URI.
+
+Further, the[Naming.md](https://github.com/OpenLineage/OpenLineage/blob/main/spec/Naming.md) has some shortcomings besides support for only databases, datalakes, and distributed file stores:
+
+- Little information, beyond Airflow and Spark, on how Jobs are named within integrations
+- Limited set of dataset names, no specific process for adding more
+- No definitions of any major terms
+- Not programmatic; reference only
+
+There also seems to be inconsistency with naming between certain integrations (at the time of this writing, the SnowflakeExtractor and Great Expectations emit different namespaces and names for Snowflake datasets. This proposal would help rectify issues that cause these discrepancies and ensure that when changes to naming happen, they are consistent across all integrations. The easiest way to do this may be to make things more programmatic, or at least have the ability to test naming changes against integrations.
+
+**Proposed implementation**
+The Naming.md file should be transitioned to a set of JSON schema files that specify the naming conditions for a particular integration or source, which can then be used to build a reference file.
```

Comment:
Can we get example of those schemas? 

I'm especially curious how we'd handle different parts of name/namespace.

Reply:
Towards the bottom of this file I show two different possibilities -- would you like a firm example?

Reply:
@denimalpaca yes, I think actual schema would help